### PR TITLE
Add microsoft logging and replace LibLog

### PR DESCRIPTION
**Pull request by jafin**
This is an extension of work started in #1086 by @lahma  Please peer review, and offer any improvements.  If later merged, suggest aborting #1086 as this would supersede that PR.

Feature supports #988 

+ Remove LibLog.
+ Added to Examples sample wireups for `Microsoft Logging` and `Serilog`. 
+ Added `SetLoggerFactory` to `QuartzConfigurator`, by default if not set, a NullLogger will be returned
+ Update all log instances to use `ILogger<T>`
+ Update log messages to MessageTemplates with template values as parameters.
+ Replaced `FailFastLoggerFactory` with `TestLoggerHelper` using MELT for log assertion.

The general approach, get logger via a service locator pattern, to inject logger in constructor appears it would need a much larger refactor, outside scope of this change.

:warning: NOTE: I have not run the integration tests, I've yet been unable to configure the required dependencies.

## TODO

Question:  `XMLSchedulingDataProcessor` takes logger in the constructor, which is different to everywhere else, perhaps this should be reverted to a service locator pattern and create a logger in the constructor for consistency?

### CODE REVIEW

**Review by lahma — CHANGES_REQUESTED** [review]
Great work @jafin ! This is quite big PR so I'm afraid I need ask a couple of things.

* this requires rebase, can you do that? (might make it easier if you squash some commits first)
* can you separate documentation changes to a separate PR that I could review after this one?

Overall I'm eager to get this into v4 branch as this is very important groundwork. Sorry it took a while for me to go start reviewing this.

**Comment by jafin** [comment]
@lahma 
- Doco has been rebased.
- Rebased on v4 branch.

Let me know if you need further changes.

File: src/Quartz.Examples/Program.cs [hunk 1]
```diff
@@ -38,8 +38,20 @@ public static async Task Main()
         {
             try
             {
-                var logRepository = log4net.LogManager.GetRepository(Assembly.GetEntryAssembly());
-                log4net.Config.XmlConfigurator.Configure(logRepository, new System.IO.FileInfo("log4net.config"));
+                var logOption = AnsiConsole.Prompt(
```
Comment (lahma) [inline src/Quartz.Examples/Program.cs:41]: Nice!

File: src/Quartz.Plugins/Plugin/History/LoggingTriggerHistoryPlugin.cs [hunk 2]
```diff
@@ -213,7 +217,7 @@ public class LoggingTriggerHistoryPlugin : ISchedulerPlugin, ITriggerListener
         /// <summary>
         /// Logger instance to use. Defaults to common logging.
```
Comment (lahma) [inline src/Quartz.Plugins/Plugin/History/LoggingTriggerHistoryPlugin.cs:218]: Seems outdated

**Review by lahma — APPROVED** [review]
Great work, thank you for finalizing this. Next step will be utilizing MS DI fully and injecting dependencies using it.

**Pull request merged**